go-controller/pkg/ovn/policy.go

```diff
@@ -1546,17 +1591,21 @@ func (oc *Controller) cleanupNetworkPolicy(np *networkPolicy) error {
 	}
 	txOkCallBack()
 
-	// Delete ingress/egress address sets
-	for _, policy := range np.ingressPolicies {
+	// Delete ingress/egress address sets, idempotent
+	for i, policy := range np.ingressPolicies {
 		err = policy.destroy()
 		if err != nil {
+			// remove deleted policies from the list
+			np.ingressPolicies = np.ingressPolicies[i:]
```

Comment:
I am not sure I understand
This slice update only happens in case of error and not on every iteration, it is supposed to remove successfully cleaned up policies, and leave the ones that are not deleted yet
It returns right after that update, so no more iterations on update slice will happen

Reply:
oops thats what I get for reviewing while :face_with_thermometer: I didn't see the if err != nil. Looks right.